text.cabal

```diff
@@ -76,13 +76,26 @@ flag developer
 flag simdutf
   description: use simdutf library, causes Data.Text.Internal.Validate.Simd to be exposed
   default: True
-  manual: True
+
+flag pure-haskell
+  description: Don't use text's standard C routines
+    When this flag is true, text will use pure Haskell variants of the
+    routines. This is not recommended except for use with GHC's JavaScript
+    backend.
+  default: False
 
 library
-  c-sources:    cbits/is_ascii.c
+  if flag(pure-haskell)
+    cpp-options: -DPURE_HASKELL
+    c-sources:  cbits/is_ascii.c
+                cbits/measure_off.c
+                cbits/utils.c
```

Comment:
```suggestion
```

Reply:
Only `reverse` is implemented in this patch, isn't it?